js/src/wrappers/anthropic.test.ts

```diff
@@ -232,4 +243,254 @@ describe("anthropic client unit tests", () => {
+    const responses = await Promise.all(
+      clients.map(async (c) => {
+        startTime = getCurrentUnixTimestamp();
+
+        const response = await c.beta.messages.create({
+          model: TEST_MODEL,
+          messages: [{ role: "user", content: "What's 4*4?" }],
+          max_tokens: 100,
+          system: "Return the result only.",
+          temperature: 0.1,
+          stream: true,
+        });
+
+        let ttft = 0;
+        let outputText = "";
+        for await (const event of response) {
+          if (ttft === 0) {
+            ttft = getCurrentUnixTimestamp() - startTime;
+          }
+          if (
+            event.type === "content_block_delta" &&
+            event.delta.type === "text_delta"
+          ) {
+            outputText += event.delta.text;
+          }
+        }
+
+        endTime = getCurrentUnixTimestamp();
+        return outputText;
+      }),
+    );
+
+    // validate that the wrapped and unwrapped clients produce the same output
+    expect(responses.length).toBe(2);
+    expect(responses[0]).toBe(responses[1]);
+
+    // validate we traced the second request
+    const spans = await backgroundLogger.drain();
+    expect(spans).toHaveLength(1);
+    const span = spans[0] as any;
+
+    expect(span).toMatchObject({
+      project_id: expect.any(String),
+      log_id: expect.any(String),
+      created: expect.any(String),
+      span_id: expect.any(String),
+      root_span_id: expect.any(String),
+      span_attributes: {
+        type: "llm",
+        name: "anthropic.messages.create",
+      },
+      metadata: {
+        model: TEST_MODEL,
+        max_tokens: 100,
+        temperature: 0.1,
+      },
+      input: [
+        { role: "user", content: "What's 4*4?" },
+        { role: "system", content: "Return the result only." },
+      ],
+      output: expect.stringContaining("16"),
+      metrics: {
+        prompt_tokens: expect.any(Number),
+        completion_tokens: expect.any(Number),
+        tokens: expect.any(Number),
+        start: expect.any(Number),
+        end: expect.any(Number),
+        time_to_first_token: expect.any(Number),
+      },
+    });
+
+    const { metrics } = span;
+    assertValidMetrics(metrics, startTime, endTime);
+  });
+
+  test("test client.beta.messages.stream", async () => {
+    let startTime: number = -1;
+    let endTime: number = -1;
+
+    // wrapped has to be last so the timing test works
+    const clients = [anthropic, client];
+    const responses = await Promise.all(
+      clients.map(async (c) => {
+        startTime = getCurrentUnixTimestamp();
+
+        const stream = c.beta.messages.stream({
+          model: TEST_MODEL,
+          messages: [{ role: "user", content: "What's 4*4?" }],
+          max_tokens: 100,
+          system: "Return the result only.",
+          temperature: 0.1,
+        });
+
+        let ttft = 0;
+        let outputText = "";
+        for await (const event of stream) {
```

Comment:
you could collect all of the chunks and could do a chunk by chunk comparison (encouraged to use the expect.any() you've been doing). maybe too detailed, though.

would it make easier for future us, though, to see how the shape changes (i.e. if i add a new attribute like reasoning)

Reply:
I like this idea